### PR TITLE
Add ownership verification steps to connection flow

### PR DESCRIPTION
**Pull request by delputnam**
#### Changes proposed in this Pull Request

In certain cases, we may want to request that users verify ownership of domains they want to connect to their site. This PR adds a flow to allow this when connecting a domain to a site with a paid plan.

When you are on a site that requires ownership verification, you will first be prompted to log in to your provider:
Screen Shot 2021-09-09 at 10 07 33 AM

In the next step, you will be asked to enter the auth code for your domain:
Screen Shot 2021-09-09 at 10 07 39 AM

Leaving the auth code field blank or entering an incorrect value will cause errors to be shown below the auth code field:
Screen Shot 2021-09-09 at 10 07 44 AM

When the correct auth code is entered, the domain will be connected and the user will be directed to the "connect a domain flow" as usual. 

#### Testing instructions

Hack the back end to return `{ ownership_verification_type: 'auth_code' }` on the `is-available` endpoint.

Select a site that is on a paid plan and go through the "Use a domain I own" to connect a domain you already own that is not mapped to WordPress.com.

Try to verify the auth code without entering anything in the input field.
Try entering the incorrect auth code in the input field.
Make sure that your domain is properly connected when you enter the correct auth code.

### CODE REVIEW

File: client/components/domains/connect-domain-step/connect-domain-step-ownership-auth-code.jsx
Comment (a8ci18n) [inline client/components/domains/connect-domain-step/connect-domain-step-ownership-auth-code.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Check my authorization code' )` *ES Score: 9*

**Review by poligilad-auto — DISMISSED** [review]
Looking good @delputnam.

1. The first step is missing some copy. I added in an additional sentence before the explanation about how to log in, to explain why they need to go through this step. "We need to confirm that you are authorized to connect this domain." ([Figma file](https://www.figma.com/file/kjzWtItHG8sIJZBJGxpWBX/Early-touch-domains?node-id=2104%3A74548))
2. For the second step, following your comment we can use "Enter authorization code" as the placeholder text and "Check my authorization code" as the CTA.

File: packages/wpcom.js/src/lib/domains.js
Comment (hambai) [inline packages/wpcom.js/src/lib/domains.js]: We already have a method called `checkAuthCode` in undocumented.js

**Review by rafaelgallani — COMMENTED** [review]
Nice, LGTM! After merging #56031, this one should be good to go code-wise 😬

**Comment by delputnam** [comment]
I've addressed @poligilad-auto 's comments in https://github.com/Automattic/wp-calypso/pull/56011/commits/7151e52123d916a932d7dc9d21d37da5eb16879f and https://github.com/Automattic/wp-calypso/pull/56011/commits/25a2d0ed4eaff99d56eba47b7add3422d545260c

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6578831

Thank you @delputnam for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6578831).

**Pull request merged**
